programs/odbc-bridge/ODBCConnectionFactory.h

```diff
@@ -60,6 +64,22 @@ class ODBCConnectionFactory final : private boost::noncopyable
         return ret;
     }
 
+    /// this check is performed only on the connection which was already successfully open before.
+    static bool needReconnect(nanodbc::connection & connection)
+    {
+        try
+        {
+            /// just_execute - execution without preparing any result object.
+            just_execute(connection, ODBC_CHECK_CONNECTION_QUERY);
+        }
+        catch (const nanodbc::database_error &)
+        {
+            return true;
```

Comment:
Well, actually it will work :D By reconnecting every time (but it will not the exception you may expect to get :) )